In [15]:
import numpy as np
import pandas as pd

In [16]:
# The statistical module used to run chi square test
import scipy.stats as stats

In [17]:
# Observed data in a (hypothetical) survey of 300 people 
observed = pd.Series([220,55,25], index=["omnivores", "carnivores", "herbivores"])
observed

omnivores     220
carnivores     55
herbivores     25
dtype: int64

In [18]:
# Create a data frame
df = pd.DataFrame([observed])
df

omnivores  carnivores  herbivores
0        220          55          25

In [19]:
# Create a data frame TRANSPOSED MIRA LA T MAYUSCULA
df = pd.DataFrame([observed]).T
df

0
omnivores   220
carnivores   55
herbivores   25

In [20]:
# Add a column whose default values are the expected values
df[1] = 100
df

0    1
omnivores   220  100
carnivores   55  100
herbivores   25  100

In [21]:
# Rename columns
df.columns = ["observed", "expected"]
df

observed  expected
omnivores        220       100
carnivores        55       100
herbivores        25       100

In [22]:
# View the data frame
df

observed  expected
omnivores        220       100
carnivores        55       100
herbivores        25       100

In [23]:
# The degree of freedom is 3-1 = 2
# With a p-value of 0.05, the confidence level is 1.00-0.05 = 0.95.
critical_value = stats.chi2.ppf(q = 0.95, df = 2)

In [24]:
# The critical value
critical_value

5.991464547107979

In [25]:
# Run the chi square test with stats.chisquare()
stats.chisquare(df['observed'], df['expected'])

Power_divergenceResult(statistic=220.5, pvalue=1.3153258948574585e-48)

### Conclusion

* Since the chi square value of 220.5 exceeds the critical value of 5.99, we conclude that the results are statistically significant.